src/archive.rs

```diff
@@ -143,7 +143,15 @@ impl<R: Read> Archive<R> {
     }
 }
 
-impl<'a> Archive<dyn Read + 'a> {
+impl<R: Read + Seek> Archive<R> {
+    /// Create a new archive using a reader that also implements `Seek`, to
+    /// allow more efficient skipping over file contents.
+    pub fn new_from_seek(obj: R) -> Archive<SeekingSkipRead<R>> {
```

Comment:
I would ideally prefer to avoid exposing `SkipRead` from this crate or the `SeekingSkipRead` type, instead making `Archive<R>` simply "work" if `R` implements `Seek`. That I don't think is natively possible without specialization, but I still think it may be possible to get somewhat close.

I'm imagining something where you'd still use `Archive::new` with seek-able readers, but insteadd of calling `entries` you'd call something like `entries_with_seek` which requires `R: Read + Seek`. The internal implementation details would still be the same as you have today with a custom trait, but the custom trait would be an implementation detail rather than a public API detail.

Would you be up for seeing if this would work?

Reply:
I really like that idea. I uploaded a new version that provides a `entries_with_seek` as you suggested and uses a different mechanism to avoid modifying the public API.